.github/workflows/cmake-superbuild/CMakeLists.txt

```diff
@@ -21,16 +37,40 @@ set(CMAKE_ARGS
         -DCMAKE_PREFIX_PATH=${DEVROOT}
         -DCMAKE_INSTALL_PREFIX=${DEVROOT}
         -DCMAKE_FIND_ROOT_PATH=${DEVROOT}
+        -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
         )
 
 set(LIBS "")
 if (NOT (ZMQ_TAG STREQUAL "system"))
-    ExternalProject_add(
-            zmq
-            CMAKE_ARGS ${CMAKE_ARGS} -DZMQ_BUILD_TESTS=OFF -DWITH_PERF_TOOL=OFF -DWITH_LIBSODIUM=OFF
-            GIT_REPOSITORY https://github.com/zeromq/libzmq.git
-            GIT_TAG ${ZMQ_TAG}
-    )
+    if (ZMQ_BUILD_WITH_CMAKE)
+        message(STATUS " build libzmq with cmake ")
+        ExternalProject_add(
```

Comment:
Have you looked at using `FetchContent_Declare` and `FetchContent_MakeAvailable` instead of `ExternalProject_add`?  I've used theses with zeromq myself.  The big benefit of FetchContent is that the targets that are fetched are available at CMake configure time rather than only at build time.  Of course you wouldn't be able to build it with autotools this way is it needed to be able to do that?

Reply:
Yes, `FetchContent` is much simpler to use.
But [CPM.cmake](https://github.com/cpm-cmake/CPM.cmake) even better than `FetchContent`.

Reply:
Ah cool.  I will take a look - thanks.